File: backend_springboot/src/main/java/com/polideportivo/backend_springboot/domain/model/BlacklistToken.java
package com.polideportivo.backend_springboot.domain.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "blacklist_token")
public class BlacklistToken {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "usuario_id", nullable = false)
    @ToString.Exclude
    private Usuario user;  // Relación con Usuario

    @Column(nullable = false)
    private String refreshToken;  // El refresh token que se ha revocado

    @CreationTimestamp
    @Column(nullable = false)
    private LocalDateTime revokeTime;  // Fecha en la que se revocó el token
}

File: backend_laravel/app/Models/BlacklistToken.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class BlacklistToken extends Model
{
    protected $table = 'blacklist_token';
    
    protected $fillable = [
        'usuario_id', 
        'refresh_token',
        'revoke_time'
    ];

    public function usuario()
    {
        return $this->belongsTo(Usuario::class, 'usuario_id'); 
    }
}
